Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -448,21 +461,38 @@ void FunctionDefinitionWriter::visit(AST::CallExpression& call)
                 m_stringBuilder.append(m_indent);
                 visit(argument);
                 m_stringBuilder.append(",\n");
-                first = false;
             }
         }
         m_stringBuilder.append(m_indent, "}");
-    } else {
-        visit(call.target());
-        m_stringBuilder.append("(");
-        for (auto& argument : call.arguments()) {
-            if (!first)
-                m_stringBuilder.append(", ");
-            visit(argument);
-            first = false;
+        return;
+    }
+
+    if (is<AST::NamedTypeName>(call.target())) {
+#define ALIAS(alias) \
+    [](FunctionDefinitionWriter* writer, AST::CallExpression& call) { \
+        writer->m_stringBuilder.append(alias); \
+        visitArguments(writer, call, 1); \
+    }
+
+        static constexpr std::pair<ComparableASCIILiteral, void(*)(FunctionDefinitionWriter*, AST::CallExpression&)> builtinMappings[] {
+            { "textureSample", [](FunctionDefinitionWriter* writer, AST::CallExpression& call) {
+                ASSERT(call.arguments().size() > 1);
+                writer->visit(call.arguments()[0]);
+                writer->m_stringBuilder.append(".sample");
+                visitArguments(writer, call, 1);
+            } },
+            { "vec4", ALIAS("float4") },
+        };
```

Comment:
The indentation doesn't seem to be right

Reply:
Do we have to use lambdas? What's wrong with a plain-old-function?

Reply:
> The indentation doesn't seem to be right

Any particular reason? The linter is happy, and I don't see anything wrong. The `#define` can't be indented or the style checker complains, and everything else is indented accordingly as far as I can tell.

> Do we have to use lambdas? What's wrong with a plain-old-function?

We certainly don't have to :) but I thought it was nicer to have everything defined in one place vs the function + the dictionary separate. Is there a reason you would prefer plain functions over lambdas?